src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -339,6 +372,549 @@ CreateInsertSelectRouterPlan(Query *originalQuery,
 
 
 /*
+ * AllRelationsJoinedOnPartitionKey expands the var equivalances gathered from
```

Comment:
@anarazel This function and this approach cannot pushdown queries that don't involve any joins but has more than 1 relations, namely set operations:

```SQL
INSERT INTO agg_events 
            (user_id) 
SELECT user_id 
FROM   ((SELECT user_id 
         FROM   raw_events_first) 
        UNION 
        (SELECT user_id 
         FROM   raw_events_first)) AS t;   
```

Do you see any viable approach to handle such queries?

Note that the following query can be pushed down since it involves join:

```SQL
INSERT INTO agg_events 
            (user_id) 
SELECT t.user_id 
FROM   ((SELECT user_id 
         FROM   raw_events_first) 
        UNION 
        (SELECT user_id 
         FROM   raw_events_first)) AS t JOIN raw_events_second ON t.user_id = raw_events_second.user_id; 
```